BridgeCareApp/BridgeCare/DataAccessLayer/DataAdapters/AttributeYearDataTableAdapter.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Data;
+using System.Data.SqlClient;
+using BridgeCare.Models;
+
+namespace BridgeCare.DataAccessLayer.DataAdapters
+{
+    /// <summary>
+    /// Populate list of AttributeByYearModel
+    /// </summary>
+    public class AttributeYearDataTable
+    {
+        private DataTable AttributesDataTable;
+        private List<AttributeByYearModel> TheAttributeYearList;
+
+        public AttributeYearDataTable(DataTable tableWithAttributes)
+        {
+            AttributesDataTable = tableWithAttributes ?? throw new ArgumentNullException(nameof(tableWithAttributes));
+
+            TheAttributeYearList =  DataTableToAttributeList(tableWithAttributes);
+        }
+
+        public List<AttributeByYearModel> Fetch() => TheAttributeYearList;
+
+
+        private bool IsDigitsOnly(string str)
+        {
+            foreach (char c in str)
```

Comment:
Just use int.TryParse to see if something is an integer.

Reply:
private bool IsNumber(string str) => int.TryParse(str, out _);

changed name to reflect how tryparse is being used and streamlined.